src/test/app/AMM_test.cpp

```diff
@@ -6862,6 +6847,37 @@ struct AMM_test : public jtx::AMMTest
         }
     }
 
+    void
+    testIssuerClawback(FeatureBitset features)
+    {
+        testcase("Issuer Clawback");
+        using namespace jtx;
+
+        // Issuer has clawback enabled
+        {
+            Env env(*this, features);
+            env.fund(XRP(1'000), gw);
+            env(fset(gw, asfAllowTrustLineClawback));
+            fund(env, gw, {alice}, XRP(1'000), {USD(1'000)}, Fund::Acct);
+            env.close();
+
+            if (!features[featureAMMClawback])
```

Comment:
should add some comments before each condition block to describe the test and expected output